bundles/automation/org.eclipse.smarthome.automation.event.test/src/test/groovy/org/eclipse/smarthome/automation/event/RuleEventTest.groovy

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * This tests events of rules
+ * 
+ * @author Benedikt Niehues - initial contribution
+ *
+ */
+class RuleEventTest extends OSGiTest{
+
+    final Logger logger = LoggerFactory.getLogger(RuleEventTest.class)
+    VolatileStorageService volatileStorageService = new VolatileStorageService()
+
+    @Before
+    void before() {
+        def itemProvider = [
+            getAll: {
+                [
+                    new SwitchItem("myMotionItem"),
+                    new SwitchItem("myPresenceItem"),
+                    new SwitchItem("myLampItem"),
+                    new SwitchItem("myMotionItem2"),
+                    new SwitchItem("myPresenceItem2"),
+                    new SwitchItem("myLampItem2")
+                ]
+            },
+            addProviderChangeListener: {},
+            removeProviderChangeListener: {},
+            allItemsChanged: {}] as ItemProvider
+        registerService(itemProvider)
+        registerService(volatileStorageService)
+
+        def autoupdateConfig = [
+            autoUpdate: { String itemName -> return true }
+
+        ] as AutoUpdateBindingConfigProvider
+        registerService(autoupdateConfig)
+    }
+
+    @Test
+    public void testRuleEvents() {
+        
+        //Registering eventSubscriber
+        def ruleEvents = []
+        
+        def ruleEventHandler = [
+            receive: {  Event e ->
+                logger.info("RuleEvent: " + e.topic)
+                    ruleEvents.add(e)
+            },
+
+            getSubscribedEventTypes: {
+                Sets.newHashSet(RuleAddedEvent.TYPE, RuleRemovedEvent.TYPE, RuleStatusInfoEvent.TYPE, RuleUpdatedEvent.TYPE)
+            },
+
+            getEventFilter:{ null }
+            ] as EventSubscriber
+        registerService(ruleEventHandler)
+        
+        //Creation of RULE
+        def triggerConfig = [itemName:"myMotionItem2"]
+        def condition1Config = [operator:"=", itemName:"myPresenceItem2", state:"ON"]
+        def condition2Config = [operator:"=", itemName:"myMotionItem2", state:"ON"]
+        def actionConfig = [itemName:"myLampItem2", command:"ON"]
+        def triggers = [
+            new Trigger("ItemStateChangeTrigger2", "ItemStateChangeTrigger", triggerConfig)
+        ]
+        def conditions = [
+            new Condition("ItemStateCondition3", "ItemStateCondition", condition1Config, null),
+            new Condition("ItemStateCondition4", "ItemStateCondition", condition2Config, null)
+        ]
+        def actions = [
+            new Action("ItemPostCommandAction2", "ItemPostCommandAction", actionConfig, null)
+        ]
+
+        def rule = new Rule("myRule21",triggers, conditions, actions, null, null)
+        rule.name="RuleEventTestingRule"
+
+        logger.info("Rule created: "+rule.getUID())
+
+        def ruleRegistry = getService(RuleRegistry)
+        ruleRegistry.add(rule)
+        ruleRegistry.setEnabled(rule.UID, true)
+
+        //TEST RULE
```

Comment:
just bringing a rule to execution, later I want to check if StatusEvents were fired...